product_pricelist_supplierinfo/models/product_product.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, tools
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def _get_supplierinfo_pricelist_price(self, rule, date=None,
+                                          quantity=None):
+        """Method for getting the price from supplier info."""
+        self.ensure_one()
+        domain = [
+            '|',
+            ('product_id', '=', self.id),
+            ('product_tmpl_id', '=', self.product_tmpl_id.id),
+        ]
+        if not rule.no_supplierinfo_min_quantity and quantity:
+            domain += [
+                '|',
+                ('min_qty', '=', False),
+                ('min_qty', '<=', quantity),
+            ]
+        if date:
+            domain += [
+                '|',
+                ('date_start', '=', False),
+                ('date_start', '<=', date),
+                '|',
+                ('date_end', '=', False),
+                ('date_end', '>=', date),
+            ]
+        # We use a different default order because we are interested in getting
+        # the price for lowest minimum quantity if no_supplierinfo_min_quantity
+        supplierinfos = self.env['product.supplierinfo'].search(
+            domain, order='sequence, min_qty, price',
+        )
+        if rule.no_supplierinfo_min_quantity:
+            price = supplierinfos[:1].price
+        else:
+            price = supplierinfos[-1:].price
```

Comment:
Why we take the last one here? To take the most expensive one?

Reply:
No, on contrary, take the price for the maximum "minimum quantity".

If you have these supplierinfo:

1. Min. qty: 1 - Price: 10
2. Min. qty: 5 - Price: 9
3. Min. qty: 10 - Price: 8

Querying for quantity=5, it will return 1 and 2. The price to apply is 2.
Querying for quantity=10, it will return 1, 2 and 3. The price to apply is 3.

Reply:
It makes sense